resources.whatwg.org/standard.css

```diff
@@ -172,40 +172,142 @@ dl.triple dt:after { content: ':'; }
+}
+.domintro > :first-child, .note > :first-child, .warning > :first-child, .example > :first-child, .XXX > :first-child, .critical > :first-child,
+.domintro > .self-link:first-child + *, .note > .self-link:first-child + *, .warning > .self-link:first-child + *, .example > .self-link:first-child + *, .XXX > .self-link:first-child + *, .critical > .self-link:first-child + * {
+  margin-top: 0;
+}
+.domintro > :last-child, .note > :last-child, .warning > :last-child, .example > :last-child, .XXX > :last-child, .critical > :last-child {
+  margin-bottom: 0;
 }
 
+/* These ones have decorations */
+.domintro, .note, .warning, .example {
+  position: relative;
+
+  /* top gets extra 0.5em (both above and below) for the decoration;
+     then we add the padding to the bottom as well for symmetry */
+  margin-top: 2.5em;
+  padding-top: 1em;
+  padding-bottom: 1em;
+}
+.domintro::before, .note::before, .warning::before, .example::before {
+  padding: 0.15em 0.25em;
+  position: absolute;
+  top: -0.8em;
+  left: -0.8em;
+  color: white;
+}
 @media (max-width: 767px) {
-  dl.domintro, .note, .warning, .example { margin-top: 1.25em; margin-left: 0; padding: 1.5em 0.5em 0.5em 1.5em; }
-  .warning { padding-top: 2em; }
-  dl.domintro::before, .note::before, .warning::before, .example::before { top: -0.3em; left: -0.3em; }
+  .domintro, .note, .warning, .example {
+    margin-top: 2em;
+    padding-top: 1.5em;
+    padding-bottom: 0.5em;
+  }
+  .domintro::before, .note::before, .warning::before, .example::before {
+    top: -0.3em;
+    left: -0.3em;
+  }
+}
+
+/* These ones can be inline <span>s */
+span.note, span.warning, span.XXX {
+  padding-top: 0;
+  padding-bottom: 0;
+}
+span.note {
+  padding-left: 3.5em;
+}
+span.warning {
+  padding-left: 6em;
+}
+span.note::before, span.warning::before {
+  left: 0;
+  top: -0.25em;
 }
 
+.domintro {
+  color: green;
+  background: #DDFFDD;
+}
+.domintro::before {
+  background: green;
+}
+.domintro dt, .domintro dt * {
+  color: black;
+}
+dl.domintro dd p {
+  margin: 0.5em 0;
+}
+dl.domintro dd p:first-child {
+  margin-top: 0;
+}
+dl.domintro dd p:last-child {
+  margin-bottom: 0;
+}
+
+.note {
+  color: green;
+  background: #DDFFDD;
+  font-style: italic;
+}
+.note::before {
+  background: green;
+}
+.note em, .note i, .note var, .note cite {
+  font-style: normal;
+}
+
+.warning {
+  color: red;
+  background: transparent;
+  font-weight: bolder;
+  font-style: italic;
+}
+.warning::before {
+  color: yellow;
+  background: red;
+}
+.warning em, .warning i, .warning var, .warning cite {
+  font-style: normal;
+}
+
+.example {
+  color: #222222;
+  background: #EEEEEE;
+}
+.example::before {
+  background: #222222;
+}
+td > .example:only-child {
+  margin: 0 0 0 0.1em;
+  padding: 0;
+}
+td > .example:only-child::before {
+  display: none;
+}
+
+
+.XXX {
+  color: #E50000;
+  background: white;
+  border: solid red;
+}
+
+.critical {
+  background: #FFFFCC;
+  border: double thick red;
+}
+
+
+
```

Comment:
Why the triple newline?

Reply:
That was my attempt at separating out this "section" of CSS. Other ideas welcome. I thought

```css
/*
 * end .domintro, .note, .warning, .example, .XXX, .critical
 */
```

was too heavy, but that's another idea. Or we could just have a normal newline.

Reply:
Current pattern is a normal newline, see e.g. self-links. Not sure if it's great though.